### PR TITLE
feat(local): add api executor functions

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

This implements the API based functions for the `local` executor.

The functions added here are:

* [GetBuild()](https://github.com/go-vela/pkg-executor/blob/29634d7c0d04fca71d141dbd56a30c67f81e6728/executor/local/api.go#L17-L25)
* [GetPipeline()](https://github.com/go-vela/pkg-executor/blob/29634d7c0d04fca71d141dbd56a30c67f81e6728/executor/local/api.go#L27-L35)
* [GetRepo()](https://github.com/go-vela/pkg-executor/blob/29634d7c0d04fca71d141dbd56a30c67f81e6728/executor/local/api.go#L37-L45)
* [CancelBuild()](https://github.com/go-vela/pkg-executor/blob/29634d7c0d04fca71d141dbd56a30c67f81e6728/executor/local/api.go#L47-L69)

### CODE REVIEW

File: executor/local/api_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package local
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestLocal_GetBuild(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/api_test.go:12]: Ignoring this 👍

File: executor/local/api_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+			t.Errorf("GetBuild returned err: %v", err)
+		}
+
+		if !reflect.DeepEqual(got, _build) {
+			t.Errorf("GetBuild is %v, want %v", got, _build)
+		}
+	}
+}
+
+func TestLocal_GetPipeline(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/api_test.go:60]: Ignoring this 👍

File: executor/local/api_test.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			t.Errorf("GetPipeline returned err: %v", err)
+		}
+
+		if !reflect.DeepEqual(got, _steps) {
+			t.Errorf("GetPipeline is %v, want %v", got, _steps)
+		}
+	}
+}
+
+func TestLocal_GetRepo(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/local/api_test.go:108]: Ignoring this 👍

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by matt-fevold — APPROVED** [review]

**Pull request merged**